torch/_inductor/codegen/cpp_prefix.h

```diff
@@ -96,6 +103,13 @@ template <typename T> void atomic_add(volatile T *addr, T offset) {
                                                std::memory_order_relaxed));
 }
 
+// Since C++20 float is supported by fetch_add, but the performance may not
+// better than compare_exchange_weak, which can be checked by microbenchmark
+// inductor_cpu_atomic.py
+template <> void atomic_add<float>(volatile float *addr, float offset) { atomic_floating_add(addr, offset); }
+template <> void atomic_add<double>(volatile double *addr, double offset) { atomic_floating_add(addr, offset); }
+template <> void atomic_add<bfloat16>(volatile bfloat16 *addr, bfloat16 offset) { atomic_floating_add(addr, offset); }
```

Comment:
Instead of specializing for floating point, can we do specialization for integral types instead since native `fetch_add` is only supported with integral types with current config? Usually, we leave the unspecialized one the most general while do specialization on things that we believe would be optimal.